api/src/main/java/gov/cms/ab2d/api/controller/BulkDataAccessAPI.java

```diff
@@ -124,14 +125,29 @@
         checkResourceTypesAndOutputFormat(resourceTypes, outputFormat);
         checkSinceTime(since);
 
-        Job job = jobService.createJob(resourceTypes, ServletUriComponentsBuilder.fromCurrentRequest().toUriString(),
-                null, outputFormat, since);
+        Job job = jobService.createJob(resourceTypes, getCurrentUrl(), null, outputFormat, since);
 
         logSuccessfulJobCreation(job);
 
         return returnStatusForJobCreation(job);
     }
 
+    private String getCurrentUrl() {
+        String request = ServletUriComponentsBuilder.fromCurrentRequest().toUriString();
+        request = request.replace(":80/", "/");
```

Comment:
I did not see `80` anywhere within the URLs, in which case we _might_ not need this replacement

Reply:
Oh we do. For some reason, at some point, it ends up creating the url https://localhost:80.

Reply:
And adding scheme doesn't seem to fix that. It's when it's generating the file names but the request comes in as http.